src/reconcile/rec_visibility.c

```diff
@@ -657,50 +657,45 @@ __rec_fill_tw_from_upd_select(
          * currently we either append the onpage value and return that, or return the tombstone
          * itself; there is no case that returns no update but sets the time window.)
          *
-         * FIXME-WT-6557: no need to check this after WT-6557 is done as the tombstone will be freed
-         * when it is written to the disk image in the previous eviction.
+         * If the tombstone is restored from the disk or the history store, the onpage value and the
+         * history store value should have been restored together. Therefore, we should not end up
+         * here.
          */
-        if (!F_ISSET(tombstone, WT_UPDATE_RESTORED_FROM_DS | WT_UPDATE_RESTORED_FROM_HS)) {
-            WT_RET(__rec_append_orig_value(session, page, tombstone, vpack));
+        WT_ASSERT_ALWAYS(session,
+          !F_ISSET(tombstone, WT_UPDATE_RESTORED_FROM_DS | WT_UPDATE_RESTORED_FROM_HS),
+          "Tombstone have been written to the disk image or history store should be accompanied by "
```

Comment:
Grammar nitpick: "Tombstone have been written" should be "A tombstone written"

Reply:
Thanks